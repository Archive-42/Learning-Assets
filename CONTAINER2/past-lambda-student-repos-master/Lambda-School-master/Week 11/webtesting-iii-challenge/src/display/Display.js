import React from 'react';

const Display = ({ closed, locked }) => {
  const closedClass = `led ${closed ? 'red-led' : 'green-led'}`;
  const lockedClass = `led ${locked ? 'red-led' : 'green-led'}`;

  return (
    <div className="display panel">
      <div className={lockedClass}>{locked ? 'Locked' : 'Unlocked'}</div>
      <div className={closedClass}>{closed ? 'Closed' : 'Open'}</div>
    </div>
  );
};

Display.defaultProps = {
  closed: false,
  locked: false,
};

export default Display;
