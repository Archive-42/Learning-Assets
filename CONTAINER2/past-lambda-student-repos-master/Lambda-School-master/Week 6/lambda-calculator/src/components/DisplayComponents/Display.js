import React from "react";

// Displays total result
const Display = () => {
  return (
    <div className="display">
      <h3>30 + 5</h3>
      <div class="display-result">
        <h2>35</h2>
      </div>
    </div>
  );
};

export default Display;
