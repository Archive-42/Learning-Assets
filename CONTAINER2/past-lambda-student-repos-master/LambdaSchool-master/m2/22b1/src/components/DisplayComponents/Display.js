import React from "react";

const Display = () => {
  return <div className="display">{/* Display any props data here */}</div>;
};
