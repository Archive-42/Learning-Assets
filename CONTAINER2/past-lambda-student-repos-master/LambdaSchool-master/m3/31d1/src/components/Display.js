import React from 'react';

/*
		â€¢ display the count of balls and strikes for the at-bat.
should be updated when the user records activity on the Dashboard component.
*/

class Display extends React.Component {
	constructor(props) {
		super(props);
		this.state = {};
	}
	render() {
		return (
			<div>
				<h2>Balls: </h2>
				<div id="balls">{this.props.balls}</div>

				<h2>Strikes: </h2>
				<div id="strikes">{this.props.strikes}</div>

				<h2>Inning: </h2>
				<div id="inning">{this.props.inning}</div>

				<h2>Home: </h2>
				<div id="runsH">Runs: {this.props.runsH}</div>
				<div id="errorsH">Errors: {this.props.errorsH}</div>
				<div id="outs">Outs: {this.props.outsH}</div>
				<h2>Away: </h2>
				<div id="runsA">Runs: {this.props.runsA}</div>
				<div id="errorsA">Errors: {this.props.errorsA}</div>
				<div id="outs">Outs: {this.props.outsA}</div>
			</div>
		);
	}
}

export default Display;
