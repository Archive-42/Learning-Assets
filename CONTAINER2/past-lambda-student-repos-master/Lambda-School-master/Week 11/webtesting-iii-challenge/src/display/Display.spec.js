import React from "react";
import * as rtl from "react-testing-library";
import Display from "./Display";

test("Testing if closed OR locked, should be red", () => {
  const { getByText } = rtl.render(<Display closed={true} locked={true} />);
  const closed = getByText(/closed/i);
  const locked = getByText(/locked/i);
  expect(locked.classList).toContain("red-led");
  expect(closed.classList).toContain("red-led");
});

test("matches snapshot", () => {
  const { container } = rtl.render(<Display />);
  expect(container).toMatchSnapshot();
});
