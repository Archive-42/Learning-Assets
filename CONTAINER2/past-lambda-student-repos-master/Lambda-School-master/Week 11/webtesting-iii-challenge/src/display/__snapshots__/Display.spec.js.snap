// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
</div>
`;
